fineract-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/Loan.java

```diff
@@ -2954,6 +2956,51 @@ private void handleDisbursementTransaction(final LocalDate disbursedOn, final Pa
 
     }
 
+    public void handleDownPayment(final BigDecimal disbursedAmount, final JsonCommand command,
+            final ScheduleGeneratorDTO scheduleGeneratorDTO) {
+        if (isAutoRepaymentForDownPaymentEnabled()) {
+            LocalDate disbursedOn = command.localDateValueOfParameterNamed(ACTUAL_DISBURSEMENT_DATE);
+            BigDecimal disbursedAmountPercentageForDownPayment = this.loanRepaymentScheduleDetail
+                    .getDisbursedAmountPercentageForDownPayment();
+            ExternalId externalId = ExternalId.empty();
+            if (TemporaryConfigurationServiceContainer.isExternalIdAutoGenerationEnabled()) {
+                externalId = ExternalId.generate();
+            }
+            Money downPaymentMoney = Money.of(getCurrency(), percentageOf(disbursedAmount, disbursedAmountPercentageForDownPayment));
+            LoanTransaction downPaymentTransaction = LoanTransaction.downPayment(getOffice(), downPaymentMoney, null, disbursedOn,
+                    externalId);
+            /**
```

Comment:
It's a minor / cosmetic issue. The compiler settings that we use creates a warning when javadoc style comment is used inside of a method body. We can just change this to block coment.